### PR TITLE
feature: support dynamic upstream with upstream is not variable

### PR DESCRIPTION
**Pull request by zengjinji**
support dynamic upstream with upstream is not variable

support proxy_pass command like that
```
proxy_pass http://upstream_name;
```

it need modif nginx source code, like nginx-1.16.0-dyups_upstream.patch file

### CODE REVIEW

**Comment by yzprofile** [comment]
Awesome !

Adding some test cases and doc would be better!

**Comment by zengjinji** [comment]
> Awesome !
> 
> Adding some test cases and doc would be better!

all have done, Please see it, thanks a lot

File: README.md [hunk 1]
```diff
@@ -54,11 +54,17 @@ file: conf/nginx.conf
         }
     }
 
-If your original config looks like this:
+If your original config looks like this, upstream is not variables:
 
     proxy_pass http://upstream_name;
 
-please replace it with:
+please appaly nginx-1.16.0-dyups_upstream.patch, like this:
```
Comment (chobits) [inline README.md]: typo: appaly -> apply

Comment (chobits) [inline README.md]: typo: appaly -> apply

Comment (zengjinji) [reply README.md]: fix, thank you

**Comment by chobits** [comment]
Two reasons that I dont like to merge this feature into master.

1. Not easy to maintain for one small feature:
      * It introduced a new patch for nginx core, which makes this project hard to maintain. We already have one patch for upstream check module. 
      * Also note that dyups core has 2000+ lines, a large C module
2. It does not fix all problems: Dynamic upstream deleting is not supported.

---

If some users like this feature, we can recommend your branch: https://github.com/zengjinji/ngx_http_dyups_module/tree/ngx_dyups_upstream_patch.

**Comment by zengjinji** [comment]
I agree with you. Indeed , have a patch is not easy to maintain.
But due to the limitations of nginx source code, if you want to support dynamic upstream with upstream is not varivable, you have to modif the source code. 

I should keep a separate branch. Make a reference for user who need to use this feature.

anyway, thanks a lot for you review and reply

**Comment by yunoasgit** [comment]
Tengine users find it difficult to use dyups module
https://github.com/alibaba/tengine/issues/1230 
It is recommended that documents be updated consistently or linked to the latest URL

**Pull request closed without merging**